### PR TITLE
Update file names 411

### PR DESCRIPTION
**Pull request by 3mma-mack**
Pull request to update output file names as part of issue #411. This includes updates to corresponding name constants and arguments in `prep_riboviz.nf`, `riboviz/workflow_r.py`, `riboviz/workflow_files.py`, and `generate_stats.figs.R`, as well as updating names in any documentations.

The file riboviz_outputs.md has been created to describe each of the output files in detail. 
Config file argument `do_pos_sp_nt_freq` has been updated to `output_metagene_normalized_profile`, with `riboviz/upgrade_config.py` updated to reflect this. Config files in example-datasets will likely need to be updated again to reflect this change.

The table of original and new names can be found [here.](https://github.com/riboviz/riboviz/issues/411#issuecomment-897720102)

### CODE REVIEW

File: docs/user/prep-riboviz-operation.md [hunk 1]
```diff
@@ -139,35 +139,41 @@ For each sample (`<SAMPLE_ID>`), intermediate files are produced in a sample-spe
...
+More details on the output files can be found at [RiboViz output files and figures](./riboviz_outputs.md).
 
 ---
 
 ## Read counts file
 
 The workflow will summarise information about the number of reads in the input files and in the output files produced at each step of the workflow. This summary is produced by scanning input, temporary and output directories and counting the number of reads (sequences) processed by specific stages of a RiboViz workflow.
 
-The read counts file, `read_counts.tsv`, is written into the output directory.
+The read counts file, `read_counts_per_file_tsv`, is written into the output directory.
```
Comment (mikej888) [inline docs/user/prep-riboviz-operation.md:176]: `read_counts_per_file_tsv` =>  `read_counts_per_file.tsv`

File: docs/user/riboviz_outputs.md [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+
+When looking at the `read_counts_by_length.pdf`, it is expected to see a peak of reads at 28-31nt. If the adapter or UMI (Unique Molecular Identifiers) sequence given in the config file does not match those used in the experiment irregularities will be present in the output files. If these sequences are incorrect a clear peak at 28-31nt in `read_counts_by_length.pdf` will not be present. In addition, very few reads may be present as they are unable to align to the annotation files so are discarded at an earlier stage of the pipeline. You can examine `read_counts_per_file.tsv` to check the number of aligned reads at each stage of the pipeline. 
+
+When looking at the `metagene_start_stop_read_counts.pdf`, it is expected to see a peak of reads upstream of the start codon, then repeating smaller peaks every 3nt. Severe disruption of the 3nt periodicity indicates a lower quality dataset and potential issues with the adapter sequence or UMI regular expression. 
+
+
+Descriptions of specific files can be located quickly using the following links:
+
+**Output files for an entire run**
+* [TPMs_all_CDS_all_samples](#tpms_all_cds_all_samplestsv)
```
Comment (mikej888) [inline docs/user/riboviz_outputs.md:15]: Change link text to 'TPMs_all_CDS_all_samples.tsv'

File: docs/user/riboviz_outputs.md [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+* [normalized_density_APEsites_per_codon.pdf](#normalized_density_APEsites_per_codonpdf)
+* [gene_position_length_counts_5start.tsv](#gene_position_length_counts_5starttsv)
+* [metagene_start_barplot_by_length.pdf](#metagene_start_barplot_by_lengthpdf)
+* [metagene_start_ribogrid_by_length.pdf](#metagene_start_ribogrid_by_lengthpdf)
+* [read_frame_per_ORF.tsv](#read_frame_per_ORFtsv)
+* [read_frame_per_ORF_filtered.tsv](#read_frame_per_ORF_filteredtsv)
+* [frame_proportions_per_ORF.pdf](#frame_proportions_per_ORFpdf)
+
+After a riboviz run, many output files are produced within the output directory.
+The output directory is specified by the parameter `dir_out` in `config.yaml`.
```
Comment (mikej888) [inline docs/user/riboviz_outputs.md:46]: Replace `config.yaml` with 'the YAML configuration file.'

File: docs/user/riboviz_outputs.md [hunk 4]
```diff
@@ -0,0 +1,396 @@
...
+There are a few output files that collect information for an entire run.
+There are many output files that are specific to each sample, which are organized into a separate subdirectory for each sample. 
+
+
+# Output files for an entire run
+
+
+## `TPMs_all_CDS_all_samples.tsv` 
+
+A tsv file with transcripts per million (tpm) for all genes from successfully processed samples. This file is produced by the script `collate_TPMs.R`. This script uses the `ORF_TPMs_and_counts.tsv` file from each processed sample and lists the tmps of each sample for each gene, allowing for comparison between samples.
```
Comment (mikej888) [inline docs/user/riboviz_outputs.md:57]: tmps => tpms

File: docs/user/riboviz_outputs.md [hunk 5]
```diff
@@ -0,0 +1,396 @@
...
+For each sample (`<SAMPLE_ID>`), intermediate files are produced in a sample-specific subdirectory (`<SAMPLE_ID>`).
+
+
+## `<SAMPLE_ID>_output_report.html` 
+
+This output report in .html format contains a provenance section for the sample, all the pdf output files produced by riboviz, and information on any plots not produced such as which files would be needed to produce plots in future runs. The HTML also includes a side bar to allow for navigation between figures. This file is produced by `AnalysisOutputs.Rmd`, which loads and creates all of the output graphs in HTML format.
+
+Example of the top of the HTML file:
+
+<img src="https://github.com/3mma-mack/Riboviz-honours/blob/main/riboviz_images/html_output.jpg" alt="html_output image" width="500"/>
```
Comment (mikej888) [inline docs/user/riboviz_outputs.md:96]: This is too brittle as if your repository goes the image link will break. Add the image to `docs/images/` then replace the URL with ``. Comment applies to all `` links in this doc.

File: docs/user/riboviz_outputs.md [hunk 6]
```diff
@@ -0,0 +1,396 @@
...
+10	2	0	0	0	0	0
+10	3	0	0	0	0	0
+10	4	0	0	0	0	0
+10	5	0	0	0	0	0
+10	6	0	0	0	0	0
+```
+
+*Potential names*
+
+read_nt_compostion, nt_freq_read_length, read_nt_freq
```
Comment (mikej888) [inline docs/user/riboviz_outputs.md:197]: Delete the "Potential names"

File: docs/user/riboviz_outputs.md [hunk 7]
```diff
@@ -0,0 +1,396 @@
...
+-20	398	5'
+-19	569	5'
+```
+
+
+## `metagene_start_stop_read_counts.pdf`
+
+A meta feature plot showing the total number of reads present within a 75nt window around the start and stop codons. The reads occurring at each position for each gene are summed to give a total for each position, then plotted. It is expected to see a large peak just upstream of the start codon, due to ribosome binding being the slow step of translation. This is typically followed by regular repeating smaller peaks, known as 3nt periodicity, as the majority of reads will map to the first nucleotide of a codon. If the expected features are not seen, then it is possible that there is a problem with annotation files, the adapter listed in the config files or the dataset used is of low quality. 
+
+Good quality 3nt periodicity plot:
```
Comment (mikej888) [inline docs/user/riboviz_outputs.md:148]: Rename to 'Good quality meta feature plot'

File: docs/user/riboviz_outputs.md [hunk 8]
```diff
@@ -0,0 +1,396 @@
...
+
+<img src="https://github.com/3mma-mack/Riboviz-honours/blob/main/riboviz_images/3nt_periodicity.jpg" alt="Good quality 3nt_periodicity plot" width="500"/>
+
+3nt periodicity plot produced using incorrect annotation files:
```
Comment (mikej888) [inline docs/user/riboviz_outputs.md:152]: Reword to 'Meta feature plot produced...'

File: docs/user/riboviz_outputs.md [hunk 9]
```diff
@@ -0,0 +1,396 @@
...
+
+*Potential names*
+
+read_nt_compostion, nt_freq_read_length, read_nt_freq
+
+## `metagene_normalized_profile_start_stop.pdf`
+
+A plot that shows the mean number of reads mapping to each position upstream and downstream of the start and stop codon for all genes. It is expected to have a peak at the start codon, with the majority of following positions having a relatively consistent mean that is comparatively low to the mean observed at the start codon. 
+
+Example pos_sp_rpf_norm_reads plot:
```
Comment (mikej888) [inline docs/user/riboviz_outputs.md:203]: Reword to 'Example mean number of reads plot:'

File: docs/user/riboviz_outputs.md [hunk 10]
```diff
@@ -0,0 +1,396 @@
...
+3	3.0604586928262		0.130589596966622	5'
+4	3.48406026456054	0.131466878328202	5'
+5	3.34745502397165	0.160793140986733	5'
+6	4.25230808548701	0.146130009657467	5'
+```
+
+
+## `ORF_TPMs_vs_features.tsv`
+
+A tsv file showing the tpm of a set of features and a value for that feature. The features are Length_log10 (log10 of the gene length); uATGs (number of upstream start codons); FE_atg (Free Energy at ATG); FE_cap (Free Energy at the cap); utr (length of the UTR); utr_gc (GC content of the UTR); and polyA (3' polyA tail on mRNA). This is produced by combining the tpms file with the features file, if a features file is provided. This is done by `generate_stats_figs.R` during step "Correlations between TPMs of genes with their sequence-based features" using the functions `CalculateSequenceBasedFeatures` and `WriteSequenceBasedFeatures`.
```
Comment (mikej888) [inline docs/user/riboviz_outputs.md:226]: Define features as a bulleted list so they're easier to read.

File: docs/user/riboviz_outputs.md [hunk 11]
```diff
@@ -0,0 +1,396 @@
...
+ORF	tpm	Feature	Value
+YAL001C	3.42470723290552	Length_log10	3.06445798922692
+YAL002W	1.78650805583587	Length_log10	3.10516942799933
+YAL003W	4043.513688639	  Length_log10	2.31386722036915
+YAL007C	105.576584188032	Length_log10	2.33243845991561
+YAL008W	11.8454807087593	Length_log10	2.29666519026153
+YAL010C	4.82007633057095	Length_log10	2.69284691927723
+```
+
+Only output if `--features-file` was defined.
```
Comment (mikej888) [inline docs/user/riboviz_outputs.md:238]: `--features-file` => `features_file`

File: docs/user/riboviz_outputs.md [hunk 12]
```diff
@@ -0,0 +1,396 @@
...
+
+## `ORF_TPMs_vs_features.pdf` 
+
+The features pdf relates the tpm value of different genes to a variety of different sequence features. This highlights any trends in feature value as the tpm changes. Values for the different features come from the features-file, if it is provided and are described above.
+
+Example features plot:
+
+<img src="https://github.com/3mma-mack/Riboviz-honours/blob/main/riboviz_images/features.jpg" alt=" features plot" width="500"/>
+
+Only output if `--features-file` was defined.
```
Comment (mikej888) [inline docs/user/riboviz_outputs.md:249]: `--features-file` => `features_file`

File: docs/user/riboviz_outputs.md [hunk 13]
```diff
@@ -0,0 +1,396 @@
...
+AA	Codon	tRNA	tAI	Microarray	RNA.seq	A	P	E
+K	AAA	7	0.431034	222273	82386	0.701009246604793	0.757271500717756	0.882163216975167
+N	AAC	10	0.615764	378101	110849	0.608843172114343	0.868064826175943	0.844584176727076
+K	AAG	14	1	        397111	83036	0.719699425589095	0.985491091356775	1.11948381450794
+N	AAT	6.4	0.27032	        241984.64 70943.36	0.566975111040301	0.786255203154437	0.76039088714112
+T	ACA	4	0.246373	105862	47598	0.933640392583143	1.007034200634	1.11922337626619
+```
+
+Only output if `--t-rna-file` and `--codon-positions-file` were defined.
+
```
Comment (mikej888) [inline docs/user/riboviz_outputs.md:284]: `--t-rna-file` => `t_rna_file`
`--codon-positions-file` => `codon_positions_file`

File: docs/user/riboviz_outputs.md [hunk 14]
```diff
@@ -0,0 +1,396 @@
...
+Q0065	26	0.0151338766006985	0.97837813474214
+Q0070	20	0.0103092783505155	0.66647646133505
+```
+
+
+## `normalized_density_APEsites_per_codon.tsv` 
+
+A tsv file showing the correlatation of different codons to features based on a provided tRNA file, which gives the Amino acids, the tRNA estimates, the tAI (tRNA Adaptation Index), Microarray values, and RNA.seq values for each codon. These are used to calculate mean ribosome-densities at the A/P/E sites for each codon.
+
+Produced by `generate_stats_figs` during step "Codon-specific ribosome densities for correlations with tRNAs" using functions `CalculateCodonSpecificRibosomeDensityTRNACorrelation` and `WriteCodonSpecificRibosomeDensityTRNACorrelation`.
```
Comment (mikej888) [inline docs/user/riboviz_outputs.md:272]: `generate_stats_figs` => `generate_stats_figs.R`

File: docs/user/riboviz_outputs.md [hunk 15]
```diff
@@ -0,0 +1,396 @@
...
+K	AAA	Microarray	222273	A	1.19938409239398
+K	AAA	Microarray	222273	P	1.2109078995196
+K	AAA	Microarray	222273	E	1.02770556029175
+K	AAA	RNA.seq	82386	A	1.19938409239398
+K	AAA	RNA.seq	82386	P	1.2109078995196
+K	AAA	RNA.seq	82386	E	1.02770556029175
+
+```
+
+Only output if `--t-rna-file` and `--codon-positions-file` were defined.
```
Comment (mikej888) [inline docs/user/riboviz_outputs.md:307]: `--t-rna-file` => `t_rna_file`
`--codon-positions-file` => `codon_positions_file`

File: docs/user/riboviz_outputs.md [hunk 16]
```diff
@@ -0,0 +1,396 @@
...
+
+## `normalized_density_APEsites_per_codon.pdf` 
+
+This plot shows a range of features and relates them to ribosome densitiy on the A, P and E sites. 4 features are shown; Microarry, RNA.seq, tAI and tRNA. These are taken from the `--t-rna-file` if provided. Each codon codon has a different value for each of these features, described generally here as a tRNA_value. These tRNA_values are plotted against the ribodensity at each site of the ribosome, showing any relationships or trends.
+
+Example normalized_density_apesites_per_codon plot:
+
+<img src="https://github.com/3mma-mack/Riboviz-honours/blob/main/riboviz_images/codon_ribodens.jpg" alt="codon_ribodens plot" width="500"/>
+
+Only output if `--t-rna-file` and `--codon-positions-file` were defined.
```
Comment (mikej888) [inline docs/user/riboviz_outputs.md:319]: `--t-rna-file` => `t_rna_file`
`--codon-positions-file` => `codon_positions_file`

File: docs/user/riboviz_outputs.md [hunk 17]
```diff
@@ -0,0 +1,396 @@
...
+
+Example metagene_start_barplot_by_length.pdf:
+
+<img src="https://github.com/3mma-mack/Riboviz-honours/blob/main/riboviz_images/startcodon_ribogridbar.jpg" alt="startcodon_ribogridbar plot" width="500"/>
+
+
+## `metagene_start_ribogrid_by_length.pdf`
+
+A meta-feature heatmap showing the number of reads occurring at positions around the start codons of genes with the y axis as read length, and the colour intensity showing read count. It is expected that each length will show a peak of reads just upstream of the start codon for all lengths, shown by an intense dark purple, and then an observable 3nt periodicity following the peak, which will be more distinct in the more common read lengths. This figure is created as part of `generate_stats_figs.R` step “Check for 3nt periodicity globally”, using data that is saved in the `3nt_periodicity.tsv` file.
```
Comment (mikej888) [inline docs/user/riboviz_outputs.md:350]: `3nt_periodicity.tsv` => `metagene_start_stop_read_counts.tsv`

File: docs/user/riboviz_outputs.md [hunk 18]
```diff
@@ -0,0 +1,396 @@
...
+gene	Ct_fr0	Ct_fr1	Ct_fr2	pval_fr0vs1	pval_fr0vs2	pval_fr0vsboth
+YAL068C	0	0	0	1	1	1
+YAL067W-A	0	0	0	1	1	1
+YAL067C	1	0	0	0.5	0.5	0.5
+YAL065C	65	10	56	0.000225406271329839	0.433008955500838	0.00465962908671896
+YAL064W-B	0	0	0	1	1	1
+YAL064C-A	0	0	0	1	1	1
+```
+
+Only output if `--asite-disp-length-file` was defined.
```
Comment (mikej888) [inline docs/user/riboviz_outputs.md:371]: `--asite-disp-length-file` => `asite_disp_length_file`

File: docs/user/riboviz_outputs.md [hunk 19]
```diff
@@ -0,0 +1,396 @@
...
+```
+gene	Ct_fr0	Ct_fr1	Ct_fr2	pval_fr0vs1	pval_fr0vs2	pval_fr0vsboth
+YAL065C	65	10	56	0.000225406271329839	0.433008955500838	0.00465962908671896
+YAL063C	859	207	944	2.71355108503869e-21	0.976833449475798	8.09866408105508e-08
+YAL060W	267	40	201	7.93480334492926e-15	0.0274846446691625	5.69179781323741e-12
+YAL059W	173	34	159	2.29896520195999e-09	0.339582970614079	4.5634250757101e-06
+YAL058W	43	4	31	3.99564191533441e-08	0.175807992651532	4.21595642555769e-06
+```
+
+Only output if `--asite-disp-length-file` was defined.
```
Comment (mikej888) [inline docs/user/riboviz_outputs.md:386]: `--asite-disp-length-file` => `asite_disp_length_file`

File: docs/user/riboviz_outputs.md [hunk 20]
```diff
@@ -0,0 +1,396 @@
...
+
+## `frame_proportions_per_ORF.pdf` 
+
+A box plot of the proportion of reads mapping to each nucleotide of a codon for each ORF, with the first codon being denoted as Frame 0, the second being Frame 1 and the 3rd being Frame 2. It is expected that the majority of reads will be mapping to the first nucleotide, Frame 0, so this box will be higher than the others. 
+
+Example 3ntframe_propbygene.pdf:
+
+<img src="https://github.com/3mma-mack/Riboviz-honours/blob/main/riboviz_images/3ntframe_propbygene.jpg" alt="3ntframe_propbygene plot" width="500"/>
+
+Only output if `--asite-disp-length-file` was defined.
```
Comment (mikej888) [inline docs/user/riboviz_outputs.md:396]: `--asite-disp-length-file` => `asite_disp_length_file`

File: docs/user/prep-riboviz-operation.md [hunk 21]
```diff
@@ -139,35 +139,41 @@ For each sample (`<SAMPLE_ID>`), intermediate files are produced in a sample-spe
...
-* `3ntframe_propbygene.pdf`: only output if `--asite-disp-length-file` was defined.
+* `metagene_start_stop_read_counts.tsv`
+* `metagene_start_stop_read_counts.pdf`
+* `gene_position_length_counts_5start.tsv`
+* `read_counts_by_length.tsv`
+* `read_counts_by_length.pdf`
+* `nt_freq_per_read_position.tsv`
+* `metagene_normalized_profile_start_stop.pdf`
+* `metagene_normalized_profile_start_stop.tsv`
+* `ORF_TPMs_vs_features.tsv`: only output if `--features-file` was defined.
```
Comment (mikej888) [inline docs/user/prep-riboviz-operation.md:150]: `--features-file` => `features_file`

File: docs/user/prep-riboviz-operation.md [hunk 22]
```diff
@@ -139,35 +139,41 @@ For each sample (`<SAMPLE_ID>`), intermediate files are produced in a sample-spe
...
+* `ORF_TPMs_vs_features.pdf`: only output if `--features-file` was defined.
```
Comment (mikej888) [inline docs/user/prep-riboviz-operation.md:151]: `--features-file` => `features_file`

File: docs/user/prep-riboviz-operation.md (hunk 21 above)
Comment (mikej888) [inline docs/user/prep-riboviz-operation.md:150]: `--features-file` => `features_file`

File: docs/user/prep-riboviz-operation.md [hunk 23]
```diff
@@ -139,35 +139,41 @@ For each sample (`<SAMPLE_ID>`), intermediate files are produced in a sample-spe
...
+* `ORF_TPMs_and_counts.tsv`
+* `normalized_density_APEsites_per_codon.tsv`: only output if `--t-rna-file` and `--codon-positions-file` were defined.
+* `normalized_density_APEsites_per_codon.pdf`: only output if `--t-rna-file` and `--codon-positions-file` were defined.
```
Comment (mikej888) [inline docs/user/prep-riboviz-operation.md:154]: `--t-rna-file` => `t_rna_file`
`--codon-positions-file` => `codon_positions_file`

File: docs/user/prep-riboviz-operation.md [hunk 24]
```diff
@@ -139,35 +139,41 @@ For each sample (`<SAMPLE_ID>`), intermediate files are produced in a sample-spe
...
+* `normalized_density_APEsites_per_codon_long.tsv` only output if `--t-rna-file` and `--codon-positions-file` were defined.
```
Comment (mikej888) [inline docs/user/prep-riboviz-operation.md:155]: `--t-rna-file` => `t_rna_file`
`--codon-positions-file` => `codon_positions_file`

File: docs/user/prep-riboviz-operation.md [hunk 25]
```diff
@@ -139,35 +139,41 @@ For each sample (`<SAMPLE_ID>`), intermediate files are produced in a sample-spe
...
+* `metagene_start_barplot_by_length.pdf`
+* `metagene_start_ribogrid_by_length.pdf`
+* `read_frame_per_ORF.tsv`: only output if `--asite-disp-length-file` was defined.
```
Comment (mikej888) [inline docs/user/prep-riboviz-operation.md:158]: `--asite-disp-length-file` => `asite_disp_length_file`

File: docs/user/prep-riboviz-operation.md [hunk 26]
```diff
@@ -139,35 +139,41 @@ For each sample (`<SAMPLE_ID>`), intermediate files are produced in a sample-spe
...
+* `read_frame_per_ORF_filtered.tsv`: only output if `--asite-disp-length-file` was defined.
```
Comment (mikej888) [inline docs/user/prep-riboviz-operation.md:159]: `--asite-disp-length-file` => `asite_disp_length_file`

File: docs/user/prep-riboviz-operation.md [hunk 27]
```diff
@@ -139,35 +139,41 @@ For each sample (`<SAMPLE_ID>`), intermediate files are produced in a sample-spe
...
+* `frame_proportions_per_ORF.pdf`: only output if `--asite-disp-length-file` was defined.
```
Comment (mikej888) [inline docs/user/prep-riboviz-operation.md:160]: `--asite-disp-length-file` => `asite_disp_length_file`

**Comment by mikej888** [comment]
@3mma-mack I reran all the tests and all is still OK. I've re-reviewed all the changes and have suggested changes above. All of these relate to consistency in naming of parameters and files.

Two additional suggestions are:

* Add link `[riboviz output files and figures](./docs/user/riboviz_outputs.md)` to `README.md` in the Usage bullet list under "Running the RiboViz Nextflow workflow".
* Combine content from `prep-riboviz-operation.md` `## Read counts file` section into the relevant section of your `riboviz_outputs.md` page then delete this section from `prep-riboviz-operation.md`. Once deleted update `A [read counts file](#read-counts-file) ...` in `prep-riboviz-operation.md` to `A read counts file ...`. (so the docs describe the read counts file in one place only, `riboviz_outputs.md`).

**Comment by 3mma-mack** [comment]
Thanks for the feedback and suggestions! I'll make those changes ASAP

**Comment by 3mma-mack** [comment]
I have made the changes suggested by @mikej888, including changing references of file requirements from the format `--t-rna-file` to the format `t_rna_file`, rewording sentences where necessary and removing a few final occurrences of the old file names from `riboviz_outputs.md`.

I have also updated `riboviz_outputs.md` to include the information from the `## read counts` section of `prep-riboviz-operation.md` and added a link to `riboviz-outputs.md` to the README file.

**Comment by mikej888** [comment]
@3mma-mack, thanks. I just committed a change to fix a broken img URL. All looks fine to me for a merge into `develop`. @FlicAnderson do you have any comments?

**Pull request merged**